backend/src/schema/resolvers/reports.js

```diff
@@ -1,62 +1,76 @@
-import uuid from 'uuid/v4'
-
 export default {
   Mutation: {
-    report: async (
-      _parent,
-      { resourceId, reasonCategory, reasonDescription },
-      { driver, _req, user },
-      _resolveInfo,
-    ) => {
-      const reportId = uuid()
+    report: async (_parent, params, { driver, user }, _resolveInfo) => {
+      let createdRelationshipWithNestedAttributes
+      const { resourceId, reasonCategory, reasonDescription } = params
       const session = driver.session()
-      const reportProperties = {
-        id: reportId,
-        createdAt: new Date().toISOString(),
-        reasonCategory,
-        reasonDescription,
-      }
-
-      const reportQueryRes = await session.run(
-        `
-          MATCH (u:User {id:$submitterId})-[:REPORTED]->(report)-[:REPORTED]->(resource {id: $resourceId}) 
-          RETURN labels(resource)[0] as label
-        `,
-        {
-          resourceId,
-          submitterId: user.id,
-        },
-      )
-      const [rep] = reportQueryRes.records.map(record => {
-        return {
-          label: record.get('label'),
-        }
+      const writeTxResultPromise = session.writeTransaction(async txc => {
+        const reportRelationshipTransactionResponse = await txc.run(
+          `
+            MATCH (submitter:User {id: $submitterId})
+            MATCH (resource {id: $resourceId})
+            WHERE resource:User OR resource:Comment OR resource:Post
+            CREATE (resource)<-[report:REPORTED {createdAt: $createdAt, reasonCategory: $reasonCategory, reasonDescription: $reasonDescription}]-(submitter)
+            RETURN report, submitter, resource, labels(resource)[0] as type
+          `,
+          {
+            resourceId,
+            submitterId: user.id,
+            createdAt: new Date().toISOString(),
+            reasonCategory,
+            reasonDescription,
+          },
+        )
+        return reportRelationshipTransactionResponse.records.map(record => ({
+          report: record.get('report'),
+          submitter: record.get('submitter'),
+          resource: record.get('resource').properties,
+          type: record.get('type'),
+        }))
       })
-
-      if (rep) {
-        throw new Error(rep.label)
+      try {
+        const txResult = await writeTxResultPromise
+        if (!txResult[0]) return null
+        const { report, submitter, resource, type } = txResult[0]
+        createdRelationshipWithNestedAttributes = {
+          ...report.properties,
+          post: null,
+          comment: null,
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-10-14T09:28:48Z" title="Monday, October 14th 2019, 11:28:48 am +02:00">Oct 14, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -87,6 +81,60 @@ export default {
           break
       }
 
+      return response
+    },
+  },
+  Query: {
+    reports: async (_parent, _params, { driver, _req, _user }, _resolveInfo) => {
```


if we are not using `req` and `user` there is no reason to destructure them and then put an `_`
this is a good practice for `_parent` cause it is the first parameter and we are not using it, but we don't have to destructure anything more than driver, if we are not using it

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-10-14T13:41:58Z" title="Monday, October 14th 2019, 3:41:58 pm +02:00">Oct 14, 2019</time>_

---


Yes, you are right. @mattwr18 
Done …